lib/ddtrace/contrib/resque/configuration/settings.rb

```diff
@@ -23,7 +23,23 @@ class Settings < Contrib::Configuration::Settings
           end
 
           option :service_name, default: Ext::SERVICE_NAME
-          option :workers, default: []
+
+          # A list Ruby worker classes to be instrumented.
+          # The value of `nil` has special semantics: it instruments all workers dynamically.
+          #
+          # TODO: 1.0: Automatic patching should be the default behavior.
+          # We should not provide this option in the future,
+          # as our integrations should always instrument all possible scenarios when feasible.
+          option :workers, default: nil do |o|
+            o.on_set do |value|
+              unless value.nil?
+                Datadog.logger.warn(
+                  "DEPRECATED: Resque integration now instruments all workers. \n" \
+                  'The `workers:` option is unnecessary and will be removed in the future.'
+                )
+              end
```

Comment:
Minor: Should we also warn when `value.empty?` E.g. if you're enabling the integration, but then not instrumenting any workers, that seems like a configuration mistake.

Reply:
At the end of day, the message is the same for folks with `[]` or `[MyWorker]`: don't configure it explicitly.

I think that the specific case for users with `[]` doesn't warrant it's own case, given that the deprecation warning gives them the correct instructions going forward.

On a side note, validation options is something that we should address regardless in the tracer as a whole.